src/daemon/rpc_command_executor.cpp

```diff
@@ -1913,136 +1924,140 @@ static std::optional<std::string_view> is_invalid_staking_address(
+  GET_LAST_BLOCK_HEADER::response res{};
 
-  // Query the latest block we've synced and check that the timestamp is sensible, issue a warning if not
-  {
-    GET_LAST_BLOCK_HEADER::response res{};
+  if (!e.invoke<GET_LAST_BLOCK_HEADER>({}, res, "Get latest block failed, unable to check sync status"))
+    return false;
 
-    if (!invoke<GET_LAST_BLOCK_HEADER>({}, res, "Get latest block failed, unable to check sync status"))
-      return false;
+  const auto& header = res.block_header;
+  const auto now = std::chrono::system_clock::now();
+  const auto block_ts = std::chrono::system_clock::from_time_t(header.timestamp);
 
-    auto const& header = res.block_header;
-    uint64_t const now = time(nullptr);
+  if (now - block_ts >= 10min)
+    tools::fail_msg_writer() << "The last block this Service Node knows about was about " << get_human_time_ago(now - block_ts)
+                             << "\nYour node is possibly desynced from the network or still syncing to the network."
+                             << "\n\nRegistering this node may result in a deregistration due to being out of sync with the network\n";
 
-    if (now >= header.timestamp)
-    {
-      uint64_t delta = now - header.timestamp;
-      if (delta > (60 * 60))
-      {
-        tools::fail_msg_writer() << "The last block this Service Node knows about was at least " << get_human_time_ago(header.timestamp, now)
-                                 << "\nYour node is possibly desynced from the network or still syncing to the network."
-                                 << "\n\nRegistering this node may result in a deregistration due to being out of date with the network\n";
-      }
-    }
+  if (block_height >= header.height + 5)
+    tools::fail_msg_writer() << "The last block this Service Node synced is " << block_height - header.height << " blocks away from the longest chain we know about."
+                               << "\n\nRegistering this node may result in a deregistration due to being out of sync with the network\n";
 
-    if (block_height >= header.height)
-    {
-      uint64_t delta = block_height - header.height;
-      if (delta > 15)
-      {
-        tools::fail_msg_writer() << "The last block this Service Node synced is " << delta << " blocks away from the longest chain we know about."
-                                 << "\n\nRegistering this node may result in a deregistration due to being out of date with the network\n";
-      }
-    }
-  }
+  return true;
+}
+
+// RAII-style class to temporarily clear categories and restore upon destruction (i.e. upon returning).
+struct clear_log_categories {
+  std::string categories;
+  clear_log_categories() { categories = mlog_get_categories(); mlog_set_categories(""); }
+  ~clear_log_categories() { mlog_set_categories(categories.c_str()); }
+};
+
+std::string highlight_money(uint64_t amount) {
+  return fmt::format("\x1b[36;1m{}\x1b[0m", cryptonote::format_money(amount));
```

Comment:
may want to do this selectively so that windows does not spew stuff it does't handle well.

Reply:
Windows isn't a supported platform for running oxend as a service node.  (Plus windows has supported these escapes, I believe, since 2019 or so).